src/app/app.controller.js

```diff
@@ -1,15 +1,22 @@
 angular.module('librarium')
   .controller('ApplicationController',
-    ['AuthService', '$state', '$translate', function (authService, $state, $translate) {
-      const self = this;
-
-      self.destroySession = function () {
-        authService.logout();
-        $state.transitionTo('root', {}, { reload: true });
-      };
-
-      self.changeLanguage = function (langKey) {
-        $translate.use(langKey);
-      };
-    }]);
+    ['AuthService', '$state', '$translate', '$stateParams', 'UsersService', 'CurrentUserService',
+      function (authService, $state, $translate, $stateParams, UsersService, CurrentUserService) {
+        const self = this;
+        self.currentPageId = $stateParams.id;
+
+        self.destroySession = function () {
+          authService.logout();
+          $state.transitionTo('root', {}, { reload: true });
+        };
+
+        self.changeLanguage = function (langKey) {
+          $translate.use(langKey);
+        };
+
+        self.currentUser = CurrentUserService.currentUser;
+
+        self.userProfile = UsersService.getUser(self.currentPageId)
```

Comment:
once the user logs in the app, we should save his info in a variable in UserService and in the localstorage. There's no need to query the user from the DB again. If there is data that is not available at the moment of login, we should query the user data in a resolve block in the route for the profile, or profile edit. Finally here instead of querying the DB we just get the value from the user stored in UserService